### ISSUE TITLE
User cannot proceed past Course Creator

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#304]
For at least one user, the course creation flow breaks at the end of the first Course Creator step: the creator goes gray, but nothing else happens and the course doesn't get created. This user experienced the problem on Chrome/OS X.

It looks like the failure happens before the course is saved to the database, as I can't find a record of it.

**Comment by ragesoss** [comment]
I'm asking the user to try a couple more things: logging out and logging back in, and trying in a different browser. Hopefully that will give us a little bit more of a clue.

**Comment by ragesoss** [comment]
One console log from a user:
```
Download the React DevTools for a better development experience: https://fb.me/react-devtools
Warning: Any use of a keyed object should be wrapped in React.addons.createFragment(object) before being passed as a child.
Received check
POST http://dashboard.wikiedu.org/courses.json 401 (Unauthorized)
Uncaught TypeError: Cannot read property 'message' of undefined
```

**Comment by ragesoss** [comment]
After we deployed a new build with raven-js integration to report client-side errors to Sentry, I asked the affected users to try again so that we could capture data from them about what went wrong. However, so far, none of them have been able to recreate the issue; as of yesterday, course page creation seems to be working for them. But if anyone hits this again, we will hopefully capture Sentry events.

**Comment by ragesoss** [comment]
We're pretty sure we got to the bottom of this, and 2cbf01e193036d279bbebdfc805955363dd955a1 should have solved it for good.

It turns out that this was caused by #298. If there is a broken course object in the database that has slug = `nil`, then the permissions check for saving a course fails unless the user is an admin. That explains why the gray screen problem happened to many people at the time when it did — around the same time as the broken courses got created — and then went away after a few days — because I deleted the broken course records.
